libqtile/backend/x11/window.py

```diff
@@ -903,21 +901,278 @@ def place(
+        elif top_bottom:
+            if up:
+                sibling = same[-1]
+                above = True
+            else:
+                sibling = same[0]
+                above = False
+
+        # There are no windows in the desired layer (should never happen) or
+        # we've moved to a new layer and are the only window in that layer
+        elif not same or (len(same) == 1 and moved != 0):
+            # Try to put it above the last window in the lower layers
+            if lower:
+                sibling = lower[-1]
+                above = True
+
+            # Or below the first window in the higher layers
+            elif higher:
+                sibling = higher[0]
+                above = False
+
+            # Don't think we should end up here but, if we do...
+            else:
+                # Put the window above the highest window if we're raising it
+                if up:
+                    sibling = stack[-1]
+                    above = True
+
+                # or below the lowest window if we're lowering the window
+                else:
+                    sibling = stack[0]
+                    above = False
+
+        else:
+            # Window has moved to a lower layer state
+            if moved < 0:
+                if self.kept_below:
+                    sibling = same[0]
+                    above = False
+                else:
+                    sibling = same[-1]
+                    above = True
+
+            # Window is in same layer state
+            elif moved == 0:
+                try:
+                    pos = same.index(self.wid)
+                except ValueError:
+                    pos = len(same) if up else 0
+                if not up:
+                    pos = max(0, pos - 1)
+                else:
+                    pos = min(pos + 1, len(same) - 1)
+                sibling = same[pos]
+                above = up
+
+            # Window is in a higher layer
+            else:
+                if self.kept_above:
+                    sibling = same[-1]
+                    above = True
+                else:
+                    sibling = same[0]
+                    above = False
+
+        #     # If we're already in the right position in the current layer, check if there's
+        #     # anything lower and, if so, place ourselves above it
+        #     # This is needed to ensure the window is restacked if there are windows in a
+        #     # lower layer
+        #     if sibling == self.wid and lower:
+        #         sibling = lower[-1]
+        #         above = True
+
+        # # If we're already in the right position in the current layer, check if there's
+        # # anything higher and, if so, place ourselves below it
+        # # This is needed to ensure the window is restacked if there are windows in a
+        # # higher layer
+        # if sibling == self.wid and higher:
+        #     sibling = higher[0]
+        #     above = False
```

Comment:
Just posting here in case you left these comment blocks in by accident

Reply:
Yup - these will get removed.